### PR TITLE
public interface: refactoring primeng

### PR DESCRIPTION
**Pull request by jma**
* Refactors the shared module.
* Refactors the holding items module.
* Refactors the user password module.
* Refactors the user profile edit module.
* Refactors the search bar module.
* Refactors the patron profile module.
* Adds layers to css styles.
* Merges public profiles applications into only one.
* All document detailed view tabs are now in angular.
* Updates dependencies.
* User profile application is not a webcomponent anymore.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

### CODE REVIEW

File: projects/public-search/src/app/patron-profile/patron-profile-requests/patron-profile-request/patron-profile-request.component.html [hunk 1]
```diff
@@ -16,69 +16,66 @@
...
+    </div>
+    <!-- ACTION BUTTONS ================================================= -->
+    <div class="md:col-2 col-12 flex justify-content-end">
+      @if (['PENDING', 'ITEM_IN_TRANSIT_FOR_PICKUP'].includes(record.metadata.state)) {
+        <p-button
+          id="request-cancel-{{ record.metadata.pid }}"
+          (onClick)="cancel()"
+          [disabled]="cancelInProgress"
+          [label]="'Cancel' | translate"
+          [icon]="!(!cancelInProgress || actionDone)? 'pi pi-spin pi-spinner': null"
```
Comment (PascalRepond) [inline projects/public-search/src/app/patron-profile/patron-profile-requests/patron-profile-request/patron-profile-request.component.html:54]: [nice-to-have enhancement] Add icon pi pi-cart-minus to this button and make it outline red?

File: projects/public-search/src/app/document-detail/request/request.component.html [hunk 2]
```diff
@@ -17,45 +17,31 @@
 @if (patron) {
   @if (!requestDialog) {
     @if (canRequest && canRequest.can) {
-      <dl id="record-request-{{ record.metadata.pid }}" class="row" [ngClass]="{'row pl-4': recordType === 'item', 'mb-3': recordType === 'holding'}">
-        @if (recordType !== 'holding') {
-          <dt class="mt-2 mb-3 col-lg-2 col-sm-3">&nbsp;</dt>
+      <p-button
```
Comment (PascalRepond) [inline projects/public-search/src/app/document-detail/request/request.component.html:20]: [nice to have enhancement] Add an icon 'pi pi-cart-arrow-down' to this button

**Review by PascalRepond — APPROVED** [review]

File: projects/public-search/src/app/patron-profile/patron-profile-requests/patron-profile-request/patron-profile-request.component.html (hunk 1 above)
Comment (jma) [reply projects/public-search/src/app/patron-profile/patron-profile-requests/patron-profile-request/patron-profile-request.component.html:54]: Sure? A cart on a cancel button? I would be better to use fontawsome, no?

File: projects/public-search/src/app/document-detail/request/request.component.html (hunk 2 above)
Comment (jma) [reply projects/public-search/src/app/document-detail/request/request.component.html:20]: Can I put the outlined for all type of requests?

Comment (PascalRepond) [reply projects/public-search/src/app/document-detail/request/request.component.html:20]: Yes

File: projects/public-search/src/app/patron-profile/patron-profile-requests/patron-profile-request/patron-profile-request.component.html (hunk 1 above)
Comment (PascalRepond) [reply projects/public-search/src/app/patron-profile/patron-profile-requests/patron-profile-request/patron-profile-request.component.html:54]: OK, we canuse pi-trash or fa-trash

**Pull request merged**